li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/common/LargeMessageHeaderValue.java

```diff
@@ -78,7 +76,7 @@ public byte getType() {
 
   public static byte[] toBytes(LargeMessageHeaderValue largeMessageHeaderValue) {
     byte[] serialized = largeMessageHeaderValue.getType() == LEGACY ? new byte[LEGACY_HEADER_SIZE]
-        : new byte[LEGACY_V2_HEADER_SIZE];
+        : new byte[LEGACY_V3_HEADER_SIZE];
```

Comment:
more strict check for the type?
LEGACY -> LEGACY_HEADER_SIZE
LEGACY_V2 -> LEGACY_V2_HEADER_SIZE
V3 -> LEGACY_V3_HEADER_SIZE

Reply:
updated